HeterogeneousCore/CUDAServices/plugins/NVProfilerService.cc

```diff
@@ -514,6 +518,10 @@ void NVProfilerService::preallocate(edm::service::SystemBounds const& bounds) {
 
   event_.resize(concurrentStreams);
   stream_modules_.resize(concurrentStreams);
+  for (auto& modulesForOneStream : stream_modules_) {
+    modulesForOneStream.resize(global_modules_.size(), nvtxInvalidRangeId);
+  }
```

Comment:
@wddgit does this mean that, by the time `preallocate()` is called, all modules have already been constructed, and `global_modules_` has its final size ?

Reply:
Yes. But there is one exception to that. In the case of a looper, it is possible for the looper to replace existing modules in loops after the first one with newly constructed modules in the end of loop transition.

I'm not sure if anyone actually uses that feature. In the CMSSW repository, I can see a Framework unit test and Fireworks with calls into the module replacement interface...